### DISCUSSION TITLE
How should we sync localPlanet.id for remote Players?

### DISCUSSION THREAD

**Discussion by sp00ktober** [NebulaModTeam/nebula#43] [Ideas]
As a great part of the game relies on the `localPlanet` and its ID variable and it might not always be possible to replicate the same functionality without it.

We could store the `localPlanet.id` of each user in some place where we can easily access it and update it when needed/required, for example in `LeavePlanet()` and `ArrivePlanet()`.

This would help me greatly on #36 and #35 and i think we should all agree on a way that suits us all rather then send that information in multiple packets when we implement our functionalities.

What do you think would be a good place to store this information? Or what do you think about this in general?

**Comment by Mushroom** [comment]
You mean store the localPlanet for each user on the host? What if multiple users are on the same planet then one leaves?

IMO, we need to just work out where every change to localPlanet happens and send it as a packet. Unless your suggestion is to compute some sort of delta, in which case that might work (like discord does it, the send a READY packet with the initial state then incremental deltas), which might also fix the potential issue of someone taking a while to load in and desyncing.

**Reply by sp00ktober** [reply]
I just meant to store the id, we can get the data out of the galaxy by using it.
Planet data must be in sync anyways but we can do that somewhere else

---

Basically what i mean is if we know the `localPlanet.id` of every player we can at least
- determine if we need to render things, play effects or whatever of him because we can compare to our own `localPlanet.id`
- get `PlanetData` of any player's planet and update states there or get information from there

if the `PlanetData` is synced across players i think this also could save bandwith as we can then just grab information out of the game itself rather than sending it over the network.

I mean the alternative would be to send the Planet ID each time it is needed for a computation to happen, in my case this would be for mined veins/trees/rocks but also for footstep sound data and when building any building on a planet.
That are three different kinds of network packets which all need to send that information. If we would sync this somewhere else all of that logic could just grab the info from there instead of getting it over the network.

**Comment by hubastard** [comment]
Yeah I think we will need to keep track of each players last known planet id. Else we won't be able to spawn the remote players at the appropriate location when they join back the same game. We will also need that information later to be able to send the planet state update only to the players that are currently on this planet.
